src/ssl.c

```diff
@@ -16609,12 +16609,45 @@ int wolfSSL_get_server_tmp_key(const WOLFSSL* ssl, WOLFSSL_EVP_PKEY** pkey)
 
 static int sanityCheckProtoVersion(WOLFSSL_CTX* ctx)
 {
-    if ((ctx->mask & WOLFSSL_OP_NO_SSLv3) &&
-            (ctx->mask & WOLFSSL_OP_NO_TLSv1) &&
-            (ctx->mask & WOLFSSL_OP_NO_TLSv1_1) &&
-            (ctx->mask & WOLFSSL_OP_NO_TLSv1_2) &&
-            (ctx->mask & WOLFSSL_OP_NO_TLSv1_3)) {
-        WOLFSSL_MSG("All TLS versions disabled");
+    int sanityConfirmed = 0;
+
+#ifndef NO_TLS
+    if (ctx->method->version.major == SSLv3_MAJOR) {
+    #ifdef WOLFSSL_ALLOW_SSLV3
+        if (!(ctx->mask & WOLFSSL_OP_NO_SSLv3)) {
+            sanityConfirmed = 1;
+        }
+    #endif
+    #ifndef NO_OLD_TLS
+        if (!(ctx->mask & WOLFSSL_OP_NO_TLSv1)) {
+            sanityConfirmed = 1;
+        }
+        if (!(ctx->mask & WOLFSSL_OP_NO_TLSv1_1)) {
+            sanityConfirmed = 1;
+        }
+    #endif
+    #ifndef WOLFSSL_NO_TLS12
+        if (!(ctx->mask & WOLFSSL_OP_NO_TLSv1_2)) {
+            sanityConfirmed = 1;
+        }
+    #endif
+    #ifdef WOLFSSL_TLS13
+        if (!(ctx->mask & WOLFSSL_OP_NO_TLSv1_3)) {
+            sanityConfirmed = 1;
+        }
+    #endif
+    }
+#endif
+#ifdef WOLFSSL_DTLS
+    if (ctx->method->version.major == DTLS_MAJOR) {
+        if (!sanityConfirmed) {
+            WOLFSSL_MSG("Only DTLS enabled");
```

Comment:
Message no longer makes sense.

Reply:
Now with `ctx->method->version.major` this message is no longer necessary.